scylladb/sphinx-scylladb-theme
# Reset top banner local storage entry every 30 days

Closes https://github.com/scylladb/sphinx-scylladb-theme/issues/639

- Splits the JS file in different modules for better maintenance.
- Sets an expiry date of 30 days for the top banner.

## How to test this PR

1. Clone this PR.
2. In `docs/conf.py`, set  `html_theme_options.hide_banner` to `"false"`.
3. On terminal window, go to the `docs` folder.
4. Run `make preview`
5. Open http://localhost:5500/. The site should render without errors.
6. Close the top banner.
7. Open the Inspector. In Chrome, right click > **Inspect**.
8. Go to **Application** > **Local Storage** tab.
9. Select **localhost:5050**.
10. You should see the entry "scylla-docs-hide-banner" with an expiry date.

![image](https://user-images.githubusercontent.com/9107969/206747879-48c474e2-fb06-40d5-adb2-6b2647ade391.png)

When a page is rendered, the theme checks[ if the current date is greater than the expiry date](https://github.com/scylladb/sphinx-scylladb-theme/pull/651/files#diff-38ac84ba1b1c772b4d6fdfeefaeb8ad0c8a3d3caf329593b0b179c7155c8cbadR21-R24) to invalidate the entry.
